yarn-project/prover-client/src/orchestrator/proving-state.ts

```diff
@@ -140,6 +143,21 @@ export class ProvingState {
     return this.rootParityInputs.findIndex(p => !p) === -1;
   }
 
+  public txHasPublicFunctions(index: number) {
+    return index >= 0 && this.txs.length > index && this.txs[index].publicKernelRequests.length;
+  }
+
+  public getNextPublicFunction(txIndex: number, nextIndex: number) {
```

Comment:
nit: just looking at the function, I don't know why this is "next" public function as opposed to just `getPublicFunction`

Reply:
Renamed